public/js/routing/router.js

```diff
@@ -0,0 +1,97 @@
+/**
+ * Copyright 2015-2016, Google, Inc.
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/* eslint-env browser */
+
+export default class Router {
+  constructor(window, shell, container) {
+    this.routes = new Set();
```

Comment:
It doesn't make sense to add the same route twice, therefore a Set. An Array wouldn't be an issue as well, since it would just match the first occurrence of that route.